apps/skilavottord/ws/src/app/modules/recyclingRequest/recyclingRequest.resolver.ts

```diff
@@ -64,13 +82,15 @@ export class RecyclingRequestResolver {
 
   @Mutation(() => RecyclingRequestResponse)
   async createSkilavottordRecyclingRequest(
+    @CurrentUser() user: User,
     @Args({ name: 'requestType', type: () => RecyclingRequestTypes })
     requestType: RecyclingRequestTypes,
     @Args('permno') permno: string,
     @Args('nameOfRequestor', { nullable: true }) name: string,
     @Args('partnerId', { nullable: true }) partnerId: string,
   ): Promise<typeof RecyclingRequestResponse> {
     return this.recyclingRequestService.createRecyclingRequest(
+      user,
```

Comment:
This solution isn't very intuitive... same checks are being done everywhere.

```typescript
const vehicle = await this.recyclingRequestService.getUsersVehicle(permno, user.nationalId)

if (!vehicle) {
  throw new NotFoundError(`Vehicle does not exist`) // or BadRequest ... or Forbidden
}

return this.recyclingRequestService.createRecyclingRequest(
  vehicle,
  requestType,
  name,
  user.partnerId
)
```


^ btw. here's another breaking issue ... this mutation shouldn't allow `partnerId` as an argument (needs to be removed) ... this would allow anyone to set any partner id.

Same with the name ... this is the user's name. Remove that one as well.



In addition ... doesn't this need check on the `requestType` ... should a citizen be able to set any requestType?